examples/migraphx/custom_op_miopen_kernel/custom_op_miopen_kernel.cpp

```diff
@@ -0,0 +1,160 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <algorithm>
+#include <hip/hip_runtime.h>
+#include <migraphx/migraphx.h>
+#include <miopen/miopen.h>
+#include <migraphx/migraphx.hpp> // MIGraphX's C++ API
+#include <numeric>
+
+#define MIGRAPHX_MIOPEN_ASSERT(x) (assert((x) == miopenStatusSuccess))
+#define MIGRAPHX_HIP_ASSERT(x) (assert((x) == hipSuccess))
+
+inline miopenTensorDescriptor_t make_miopen_tensor(const migraphx::shape& s, bool pack = false)
+{
+    // TODO: normalize_standard shape for scalar
+    miopenTensorDescriptor_t t;
+    MIGRAPHX_MIOPEN_ASSERT(miopenCreateTensorDescriptor(&t));
+    // Convert to ints
+    auto s_lens = s.lengths();
+    std::vector<int> lens(s_lens.begin(), s_lens.end());
+    auto s_strides = s.strides();
+    std::vector<int> strides(s_strides.begin(), s_strides.end());
+    miopenDataType_t d;
+    if(s.type() == migraphx_shape_float_type)
+        d = miopenFloat;
+    else if(s.type() == migraphx_shape_half_type)
+        d = miopenHalf;
+    else if(s.type() == migraphx_shape_int32_type)
+        d = miopenInt32;
+    else if(s.type() == migraphx_shape_int8_type)
+    {
+        if(pack)
+        {
+            // update the lens and corresponding strides
+            d          = miopenInt8x4;
+            lens[1]    = ((lens[1] + 3) / 4) * 4;
+            strides[0] = strides[1] * lens[1];
+        }
+        else
+        {
+            d = miopenInt8;
+        }
+    }
+    else
+    {
+        throw("MAKE_TENSOR: unsupported type");
+    }
+    miopenSetTensorDescriptor(t, d, s_lens.size(), lens.data(), strides.data());
+    return t;
+}
+
+inline auto make_miopen_handle(migraphx::context& ctx)
+{
+    MIGRAPHX_HIP_ASSERT(hipSetDevice(0));
+    auto* stream = ctx.get_queue<hipStream_t>();
+    miopenHandle_t out;
+    MIGRAPHX_MIOPEN_ASSERT(miopenCreateWithStream(&out, stream));
+    return out;
+}
+
+inline auto make_activation_descriptor(miopenActivationMode_t mode,
+                                       double alpha = 0,
+                                       double beta  = 0,
+                                       double gamma = 0)
+{
+    miopenActivationDescriptor_t ad;
+    MIGRAPHX_MIOPEN_ASSERT(miopenCreateActivationDescriptor(&ad));
+    miopenSetActivationDescriptor(ad, mode, alpha, beta, gamma);
+    return ad;
+}
+
+struct abs_custom_op final : migraphx::experimental_custom_op_base
+{
+    virtual std::string name() const override { return "abs_custom_op"; }
+    virtual migraphx::argument compute(migraphx::context ctx,
+                                       migraphx::shape output_shape,
+                                       migraphx::arguments args) const override
+    {
+        float alpha = 1;
+        float beta  = 0;
+        auto y_desc = make_miopen_tensor(output_shape);
```

Comment:
We should also add a comment to explain this step of making miopen tensors.

Reply:
Added.